### PR TITLE
Open audio/image input stream only when queue is ready

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

Follow up from #8941 . Will only send data for the stream when the queue has a worker available. In between joining the queue and the queue being ready, the component is notified that it is in a "waiting" state so that the UI can update accordingly.

Webcam example
image_in_out

Microphone example:
stream_wait_mic

Added `Record`, `Stop` text to the webcam record button as suggested in an internal discussion.

A demo to test with

```python
import gradio as gr

import gradio as gr
import numpy as np
import cv2


def transform_cv2(frame, transform):
    if transform == "cartoon":
        # prepare color
        img_color = cv2.pyrDown(cv2.pyrDown(frame))
        for _ in range(6):
            img_color = cv2.bilateralFilter(img_color, 9, 9, 7)
        img_color = cv2.pyrUp(cv2.pyrUp(img_color))

        # prepare edges
        img_edges = cv2.cvtColor(frame, cv2.COLOR_RGB2GRAY)
        img_edges = cv2.adaptiveThreshold(
            cv2.medianBlur(img_edges, 7),
            255,
            cv2.ADAPTIVE_THRESH_MEAN_C,
            cv2.THRESH_BINARY,
            9,
            2,
        )
        img_edges = cv2.cvtColor(img_edges, cv2.COLOR_GRAY2RGB)
        # combine color and edges
        img = cv2.bitwise_and(img_color, img_edges)
        return img
    elif transform == "edges":
        # perform edge detection
        img = cv2.cvtColor(cv2.Canny(frame, 100, 200), cv2.COLOR_GRAY2BGR)
        return img
    else:
        return np.flipud(frame)


with gr.Blocks() as demo:
    gr.Markdown("# New Image Streaming In/Out 📸")
    with gr.Row():
        with gr.Column():
            transform = gr.Dropdown(choices=["cartoon", "edges", "flip"],
                                    value="flip", label="Transformation")
            input_img = gr.Image(sources=["webcam"], type="numpy")
        with gr.Column():
            output_img = gr.Image(format="jpeg")
        dep = input_img.stream(transform_cv2, [input_img, transform], [output_img], time_limit=30, stream_every=0.1)

if __name__ == "__main__":
    demo.launch()
```
## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: js/core/src/lang/en.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
 		"record": "Record",
 		"no_microphone": "No microphone found",
 		"pause": "Pause",
-		"play": "Play"
+		"play": "Play",
```
Comment (freddyaboulton) [inline js/core/src/lang/en.json:19]: Will update the others before merging

**Comment by abidlabs** [comment]
@freddyaboulton I just tested the webcam demo:

```
import gradio as gr

import gradio as gr
import numpy as np
import cv2


def transform_cv2(frame, transform):
    if transform == "cartoon":
        # prepare color
        img_color = cv2.pyrDown(cv2.pyrDown(frame))
        for _ in range(6):
            img_color = cv2.bilateralFilter(img_color, 9, 9, 7)
        img_color = cv2.pyrUp(cv2.pyrUp(img_color))

        # prepare edges
        img_edges = cv2.cvtColor(frame, cv2.COLOR_RGB2GRAY)
        img_edges = cv2.adaptiveThreshold(
            cv2.medianBlur(img_edges, 7),
            255,
            cv2.ADAPTIVE_THRESH_MEAN_C,
            cv2.THRESH_BINARY,
            9,
            2,
        )
        img_edges = cv2.cvtColor(img_edges, cv2.COLOR_GRAY2RGB)
        # combine color and edges
        img = cv2.bitwise_and(img_color, img_edges)
        return img
    elif transform == "edges":
        # perform edge detection
        img = cv2.cvtColor(cv2.Canny(frame, 100, 200), cv2.COLOR_GRAY2BGR)
        return img
    else:
        return np.flipud(frame)


with gr.Blocks() as demo:
    gr.Markdown("# New Image Streaming In/Out 📸")
    with gr.Row():
        with gr.Column():
            transform = gr.Dropdown(choices=["cartoon", "edges", "flip"],
                                    value="flip", label="Transformation")
            input_img = gr.Image(sources=["webcam"], type="numpy")
        with gr.Column():
            output_img = gr.Image(format="jpeg")
        dep = input_img.stream(transform_cv2, [input_img, transform], [output_img], time_limit=10, stream_every=0.1)

if __name__ == "__main__":
    demo.launch()
```

and I'm seeing this behavior:
* In the first run, the output never changes (message shows I'm always in the queue even when it should be my turn)
* then, it just shows the queue size constantly increasing:


https://github.com/user-attachments/assets/0753f7cf-2250-4726-ab22-503cab788d93

**Comment by freddyaboulton** [comment]
@abidlabs - thanks for testing! Sorry about that, I refactored some code in the pr and it didn't work as I thought it would lol. Should be fixed now.

**Review by abidlabs — APPROVED** [review]
Brilliant! Tested with lots of demos and works great @freddyaboulton

**Comment by freddyaboulton** [comment]
Thank you @abidlabs !

**Pull request merged**